tokio/src/net/udp.rs

```diff
@@ -253,6 +253,78 @@ impl UdpSocket {
         }
     }
 
+    /// Sets the size of the UDP send buffer on this socket.
+    ///
+    /// On most operating systems, this sets the `SO_SNDBUF` socket option.
+    pub fn set_send_buffer_size(&self, size: u32) -> io::Result<()> {
+        self.to_socket().set_send_buffer_size(size as usize)
+    }
```

Comment:
Why the casts?

Reply:
- socket2's [set_send_buffer_size](https://docs.rs/socket2/latest/socket2/struct.Socket.html#method.set_send_buffer_size) uses usize.
- tokio::net::TcpSocket's [set_send_buffer_size](https://docs.rs/tokio/latest/tokio/net/struct.TcpSocket.html#method.set_send_buffer_size) uses u32.

This PR (tokio::net:: UdpSocket) uses u32 for consistency with TcpSocket's methods, so we need casts here.

And as I said in https://github.com/tokio-rs/tokio/pull/4270#discussion_r754974978, these casts are fine.

> - set_send_buffer_size,set_recv_buffer_size (cast u32 as usize): on tokio, [usize is always at least 32 bits](https://github.com/tokio-rs/tokio/blob/cf3206842c0d94ecdaaeb421a58b1c963b627c3d/tokio/src/lib.rs#L357-L368).
> - send_buffer_size,recv_buffer_size (cast usize as u32): socket2 gets these values as  c_int ([always i32](https://grep.app/search?q=type%20c_int%20%3D&filter[repo][0]=rust-lang/libc)) and then cast them to usize ([1](https://github.com/rust-lang/socket2/blob/cddb67b8534a46a372beb9fd63c5ce8d1473cbda/src/socket.rs#L857-L858), [2](https://github.com/rust-lang/socket2/blob/cddb67b8534a46a372beb9fd63c5ce8d1473cbda/src/socket.rs#L928-L929)).

Reply:
Thanks.